ir/memory.h

```diff
@@ -105,15 +113,12 @@ class Memory {
 
   smt::expr blocks_val; // array: (bid, offset) -> Byte
   smt::expr blocks_liveness; // array: bid -> bool
-  smt::expr blocks_kind; // array: bid -> uint(1bit), 1 if heap, 0 otherwise
-  smt::expr blocks_readonly; // array: bid -> bool, true if readonly
 
   std::string mkName(const char *str, bool src) const;
   std::string mkName(const char *str) const;
 
-  smt::expr mk_val_array(const char *name) const;
+  smt::expr mk_val_array() const;
   smt::expr mk_liveness_uf() const;
```

Comment:
As mentioned in this PR text, liveness of a memory block is not constant during execution due to the existence of free() call.
Does this mean declaring liveness as UF has no benefit than having an array? Then, I'd like to change the name of this to `mk_liveness_array`.

Reply:
right, liveness has to be an array. Renaming sounds good! :)